scripts/visualise_results.py

```diff
@@ -7,37 +7,59 @@
 
 PERCENTILES_TO_GRAPH = (50, 90, 95, 99)
 
+ADDITIONAL_METRICS_TO_GRAPH = ("PDF", "Session")
+
 
 class GraphGenerationFailed(Exception):
     pass
 
 
-def plot_data(df, number_of_days_to_plot):
-    try:
-        plt.style.use('fast')
+def plot_data(dataframes, number_of_days_to_plot):
+    plt.style.use('fast')
+    fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15, 6))
 
+    for i in range(len(dataframes)):
+        plt.subplot(1, 2, i + 1)
         if (
             number_of_days_to_plot and number_of_days_to_plot <= 45
         ):  # To make the chart still easily digestible
-            df.plot.line(marker="o", markersize=8)
+            dataframes[i].plot.line(marker="o", markersize=8, ax=axes[i])
             plt.grid(True, axis="both", alpha=0.3)
         else:
-            df.plot.line()
+            dataframes[i].plot.line(ax=axes[0])
 
         plt.margins(0.03, 0.07)
-        plt.legend(frameon=True, prop={"size": 17})
-        plt.xticks(df.index, df["DATE"], size="small", rotation=90)
+        plt.legend(frameon=True, prop={"size": 10})
+        plt.xticks(
+            dataframes[i].index, dataframes[i]["DATE"], size="small", rotation=90
+        )
         plt.yticks(size="small")
         plt.ylabel("Average Response Time (ms)")
         plt.xlabel("Run Date (YYYY-MM-DD)", labelpad=13)
 
-        plt.savefig('performance_graph.png', bbox_inches="tight")
-        print("Graph saved as performance_graph.png")
+
+def create_graph(dataframes, number_of_days_to_plot, filename):
+    try:
+        plot_data(dataframes, number_of_days_to_plot)
+        plt.savefig(filename, bbox_inches="tight")
+        print("Graph saved as", filename)
     except Exception as e:
         raise GraphGenerationFailed from e
 
 
-def get_data_frame(results):
+def create_dataframe(result_fields, values_to_plot, prefix=True):
+    if prefix:
```

Comment:
Is this referring to the suffix rather than a prefix? I think we can get rid of this if and the passing of the var if we create a percentiles_to_plot array with the `th` included (or something similar)

Reply:
Updated 👍